jerry-core/vm/vm.c

```diff
@@ -1829,6 +1829,64 @@ vm_loop (vm_frame_ctx_t *frame_ctx_p) /**< frame context */
 
           goto free_left_value;
         }
+        case VM_OC_SET_FUNCTION_NAME:
+        {
+          char *prefix_p = NULL;
+          lit_utf8_size_t prefix_size = 0;
+          if (opcode != CBC_EXT_SET_FUNCTION_NAME)
+          {
+            ecma_string_t *prop_name_p = ecma_op_to_prop_name (stack_top_p[-2]);
+
+            if (JERRY_UNLIKELY (prop_name_p == NULL))
+            {
+              result = ECMA_VALUE_ERROR;
+              goto error;
+            }
+
+            ecma_free_value (stack_top_p[-2]);
+            ecma_ref_ecma_string (prop_name_p);
+            right_value = ecma_make_prop_name_value (prop_name_p);
+            stack_top_p[-2] = right_value;
+
+            if (opcode != CBC_EXT_SET_COMPUTED_FUNCTION_NAME)
+            {
+              JERRY_ASSERT (opcode == CBC_EXT_SET_COMPUTED_GETTER_NAME
+                            || opcode == CBC_EXT_SET_COMPUTED_SETTER_NAME);
+              prefix_p = (opcode == CBC_EXT_SET_COMPUTED_GETTER_NAME) ? "get " : "set ";
+              prefix_size = 4;
+            }
+
+            left_value = stack_top_p[-1];
+          }
+          else
+          {
+            right_value = left_value;
+            left_value = stack_top_p[-1];
+          }
+
+          ecma_object_t *func_obj_p = ecma_get_object_from_value (left_value);
+          ecma_property_value_t *value_p;
+
+          if (ecma_find_named_property (func_obj_p, ecma_get_magic_string (LIT_MAGIC_STRING_NAME)) != NULL)
```

Comment:
How likely is this case? If is likely maybe we could move the whole check at the top of the case block, and user the `stack_top_p[-1]` directly as the first arg?

Reply:
Unfortunately the current order cannot be changed since the ToPropName call may trigger an accessor which sets a name property to the object.